src/reverse_complement.rs

```diff
@@ -115,92 +91,108 @@ impl<'a> Iterator for MutDnaSeqs<'a> {
+        let (right, y) = mem::replace(&mut self.right, &mut []).split_at_mut(len - n);
+        self.left = left;
+        self.right = right;
+        Some((x, y))
+    }
+}
+
 /// Length of a normal line without the terminating \n.
 const LINE_LEN: usize = 60;
 
 /// Compute the reverse complement.
 fn reverse_complement(seq: &mut [u8], tables: &Tables) {
     let len = seq.len() - 1;
     let seq = &mut seq[..len];// Drop the last newline
+
+    // Move newlines so the reversed text is wrapped correctly.
     let off = LINE_LEN - len % (LINE_LEN + 1);
     let mut i = LINE_LEN;
     while i < len {
         unsafe {
-            copy(seq.as_ptr().offset((i - off) as isize),
-                 seq.as_mut_ptr().offset((i - off + 1) as isize), off);
+            ptr::copy(seq.as_ptr().offset((i - off) as isize),
+                      seq.as_mut_ptr().offset((i - off + 1) as isize), off);
             *seq.get_unchecked_mut(i - off) = b'\n';
         }
         i += LINE_LEN + 1;
     }
 
     let div = len / 4;
     let rem = len % 4;
-    unsafe {
-        let mut left = seq.as_mut_ptr() as *mut u16;
-        // This is slow if len % 2 != 0 but still faster than bytewise operations.
-        let mut right = seq.as_mut_ptr().offset(len as isize - 2) as *mut u16;
-        let end = left.offset(div as isize);
-        while left != end {
-            let tmp = tables.cpl16(*left);
-            *left = tables.cpl16(*right);
-            *right = tmp;
-            left = left.offset(1);
-            right = right.offset(-1);
-        }
 
-        let end = end as *mut u8;
-        match rem {
-            1 => *end = tables.cpl8(*end),
-            2 => {
-                let tmp = tables.cpl8(*end);
-                *end = tables.cpl8(*end.offset(1));
-                *end.offset(1) = tmp;
-            },
-            3 => {
-                *end.offset(1) = tables.cpl8(*end.offset(1));
-                let tmp = tables.cpl8(*end);
-                *end = tables.cpl8(*end.offset(2));
-                *end.offset(2) = tmp;
-            },
-            _ => { },
+    let thread_count = num_cpus::get();
+    let chunk_size = (div + thread_count - 1) / thread_count;
+    crossbeam::scope(|scope| {
+        let (left, right);
+        unsafe {
+            let p = seq.as_mut_ptr();
+            left = slice::from_raw_parts_mut(p as *mut u16, div);
+            // This is slow if len % 2 != 0 but still faster than bytewise operations.
+            let q = p.offset((div * 2 + rem) as isize);
+            right = slice::from_raw_parts_mut(q as *mut u16, div);
+        };
+        for (left, right) in double_chunk(chunk_size, left, right) {
+            scope.spawn(move || {
+                let len = left.len();
+                for i in 0..len {
+                    let tmp = tables.cpl16(left[i]);
+                    left[i] = tables.cpl16(right[len - 1 - i]);
+                    right[len - 1 - i] = tmp;
+                }
```

Comment:
> Did you tried something like `for (x, y) in left.iter_mut().rev().zip(&mut right)`?

I tried that earlier and thought it slowed down the code, but I think I messed up the measurements. Tried it again and it works fine.